_posts/2021-03-23-solidity-0.8.3-release-announcement.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.3 Release Announcement'
+date: '2021-03-23'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.3](https://github.com/ethereum/solidity/releases/tag/v0.8.3) fixes the [Solidity Optimizer Keccak Caching Bug](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/), which is present in all prior versions of Solidity, and includes two changes to the optimizer (improve storage writes of small types and replace multiplications and divisions with powers of two by shifts).
+
```

Comment:
The improvement will make writing packed structs more gas efficient.  For example, in the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:

```solidity
struct S {
    address a;
    uint96 b;
}

contract C {
    S s;
    function f(uint96 _b) public {
        s.a = msg.sender;
        s.b = _b;
    }
}
```


-----

In the EVM, the shift operations (`shl`, `shr`, `sar`) are cheaper than the arithmetic operations `mul` and `div`. We added an optimization rule that converts multiplication and division by powers of two into `shl` and `shr` respectively. On top of the small gas saving from this simple rule, the rule also chains well with other shift operations. For example:  `shl(10, mul(x, 256)) -> shl(10, shl(8, x)) -> shl(18, x)`.

Reply:
added this